fe/src/main/java/org/apache/doris/load/loadv2/dpp/BitmapUnion.java

```diff
@@ -0,0 +1,129 @@
+
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.expressions.MutableAggregationBuffer;
+import org.apache.spark.sql.expressions.UserDefinedAggregateFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.roaringbitmap.RoaringBitmap;
+
+import java.io.IOException;
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.DataInputStream;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BitmapUnion extends UserDefinedAggregateFunction {
+    private StructType inputSchema;
+    private StructType bufferSchema;
+
+    public BitmapUnion() {
+        List<StructField> inputFields = new ArrayList<>();
+        inputFields.add(DataTypes.createStructField("str", DataTypes.StringType,true));
+        inputSchema = DataTypes.createStructType(inputFields);
+
+        List<StructField> bufferFields = new ArrayList<>();
+        bufferFields.add(DataTypes.createStructField("bitmap", DataTypes.BinaryType,true));
+        bufferSchema = DataTypes.createStructType(bufferFields);
+    }
+
+    @Override
+    public StructType inputSchema() {
+        return inputSchema;
+    }
+
+    @Override
+    public StructType bufferSchema() {
+        return bufferSchema;
+    }
+
+    @Override
+    public DataType dataType() {
+        return DataTypes.StringType;
+    }
+
+    @Override
+    public boolean deterministic() {
+        return true;
+    }
+
+    @Override
+    public void initialize(MutableAggregationBuffer buffer) {
+        RoaringBitmap roaringBitmap = new RoaringBitmap();
+        roaringBitmap.runOptimize();
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        try {
+            roaringBitmap.serialize(new DataOutputStream(bos));
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+        buffer.update(0, bos.toByteArray());
+    }
+
+    @Override
+    public void update(MutableAggregationBuffer buffer, Row input) {
+        // here maybe there is performance problems
+        if (!input.isNullAt(0)) {
+            String value = input.getString(0);
+            int id = Integer.parseInt(value);
+            Object bitmapObj = buffer.get(0);
+            byte[] bitmapBytes = (byte[])bitmapObj;
+            RoaringBitmap bitmap = new RoaringBitmap();
+            try {
+                bitmap.deserialize(new DataInputStream(new ByteArrayInputStream(bitmapBytes)));
+                bitmap.add(id);
+                bitmap.runOptimize();
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                bitmap.serialize(new DataOutputStream(bos));
+                buffer.update(0, bos.toByteArray());
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
prefer using log instead of print to std

Reply:
done